pandas/_typing.py

```diff
@@ -83,8 +84,13 @@
     # Name "npt._ArrayLikeInt_co" is not defined  [name-defined]
     NumpySorter = Optional[npt._ArrayLikeInt_co]  # type: ignore[name-defined]
 
+    if sys.version_info >= (3, 11):
+        from typing import Self
+    else:
+        from typing_extensions import Self  # pyright: reportUnusedImport = false
```

Comment:
I guess you could make Self equivalent to Any here. IIRC we did something similar before to enable the latest typing features

Reply:
We type check using a single version of Python. There are capabilities in mypy to set the Python version to check against regardless of the version used. We don't do that.

So i'm not sure when this import is actually needed.